src/plugins/intel_cpu/src/graph_optimizer.cpp

```diff
@@ -1996,7 +1996,15 @@ void GraphOptimizer::FusePerformedAsScaleShiftAndFakeQuantize(Graph &graph) {
 
         const auto &outputShape = child->getOutputShapeAtPort(0);
         VectorDims outputDims = outputShape.getDims();
-        const auto channelPos = parent->getParentEdgeAt(0)->getParent()->getFusingAxis();
+
+        // We need to compute explicitly port with unfolded parent,
+        // because there is no guarantee, that the order of operands will be invariant
+        // (i.e. zero) after all transformations, which may cause wrong channel-dim in
+        // [Const-Schift -> Add <- Mul] topology with constant-folded schift,
+        // (Const node return 1 by default as channel dim.)
+        // Look into FQScaleshiftWithConstantShift test
+        const auto nonConstPort = (parent->getParentEdgeAt(0)->getParent()->isConstant() ? 1 : 0);
```

Comment:
The FC node is built on the assumption that the constant port is always 1.
https://github.com/openvinotoolkit/openvino/blob/c03fa26e3e0516a553f7e102cefa37217e4b4554/src/plugins/intel_cpu/src/node.cpp#L871
Thus, if we have a situation when this is not the case, we are possibly in danger. Thus we need to spot and fix the reason, on which the constant may be located on port 0.